lib/dashboard/DynamicDashboardFactory.ts

```diff
@@ -0,0 +1,126 @@
+import { Duration } from "aws-cdk-lib";
+import {
+  Dashboard,
+  DashboardProps,
+  PeriodOverride,
+} from "aws-cdk-lib/aws-cloudwatch";
+import { Construct } from "constructs";
+
+import { BitmapDashboard } from "./BitmapDashboard";
+import { DashboardRenderingPreference } from "./DashboardRenderingPreference";
+import { DashboardWithBitmapCopy } from "./DashboardWithBitmapCopy";
+import { DefaultDashboards } from "./DefaultDashboardFactory";
+import { IDynamicDashboardSegment } from "./DynamicDashboardSegment";
+import { IDynamicDashboardFactory } from "./IDynamicDashboardFactory";
+
+export interface DynamicDashboardConfiguration {
+  /**
+   * Name of the dashboard. Full dashboard name will take the form of:
```

Comment:
Should document here about the default name restriction.

Reply:
Fixed.